File: platform/android/MapLibreAndroid/src/opengl/java/org/maplibre/android/module/loader/LibraryLoaderProviderImpl.java
package org.maplibre.android.module.loader;

import org.maplibre.android.LibraryLoader;
import org.maplibre.android.LibraryLoaderProvider;

/**
 * OpenGL flavor: loads the single-backend libmaplibre.so via System.loadLibrary.
 */
public class LibraryLoaderProviderImpl implements LibraryLoaderProvider {

  @Override
  public LibraryLoader getDefaultLibraryLoader() {
    return new SystemLibraryLoader();
  }

  private static class SystemLibraryLoader extends LibraryLoader {
    @Override
    public void load(String name) {
      System.loadLibrary(name);
    }
  }
}

File: platform/android/MapLibreAndroid/src/opengl/java/org/maplibre/android/module/loader/LibraryLoaderProviderImpl.kt
package org.maplibre.android.module.loader

import org.maplibre.android.LibraryLoader
import org.maplibre.android.LibraryLoaderProvider

/**
 * OpenGL flavor: loads the single-backend libmaplibre.so via System.loadLibrary.
 */
class LibraryLoaderProviderImpl : LibraryLoaderProvider {
    override fun getDefaultLibraryLoader(): LibraryLoader = SystemLibraryLoader()

    private class SystemLibraryLoader : LibraryLoader() {
        override fun load(name: String) {
            System.loadLibrary(name)
        }
    }
}
